SPAdes 4.0.0

kovcsboti opened the pull request:
The Docker image have been done using personal docker account, so a few variables should be changed.

gpetho commented:
@kovcsboti Please fix the issues noted in the Sourcery review.
Please also check whether alpine as base image would work instead of ubuntu. Obviously the package manager needs to be adjusted, possibly a few other things as well, ChatGPT can do that for you automatically. If it works, use alpine, ubuntu usually just wastes space. Obviously if it's too much of a hassle to install SPAdes on alpine because of dependencies that were not installed specifically since they are already part of the ubuntu base image, and it's not possible or practical to install these, then just keep ubuntu.

kovcsboti commented:
@gpetho  

- [X] I changed the Docker user to "deptmetagenom". I'm not sure, how usefull would it be to make it a variable if we want to use it for the metagenome department
- [X] The test reference have been changed to use relative path.
- [X] Resolved the FP_VERSION issue, changed it to SPADES_VER.
- [x] The PR description ammended by Sourcery, basically covers the main point.

Originally I tried to use Alpine and with the help of ChatGPT I managed to run the SPAdes tes, however it could not be completed it gave errors. Some of these could be rectified with more dependencies, however after a certain point ChatGPT simply offered to downgrade SPAdes to 3.15.5  .
